topdown/crypto.go

```diff
@@ -41,3 +42,67 @@ func builtinCryptoX509ParseCertificates(a ast.Value) (ast.Value, error) {
 func init() {
 	RegisterFunctionalBuiltin1(ast.CryptoX509ParseCertificates.Name, builtinCryptoX509ParseCertificates)
 }
+
+// createRootCAs creates a new Cert Pool from scratch or adds to a copy of System Certs
+func createRootCAs(tlsCaCertFile string, tlsCaCertEnvVar []byte, tlsUseSystemCerts bool) (*x509.CertPool, error) {
+
+	var newRootCAs *x509.CertPool
+
+	if tlsUseSystemCerts {
+		systemCertPool, err := x509.SystemCertPool()
+		if systemCertPool == nil {
+			return nil, err
+		}
+		newRootCAs = systemCertPool
+	} else {
+		newRootCAs = x509.NewCertPool()
+	}
+
+	if len(tlsCaCertFile) > 0 {
+		// Append our cert to the system pool
+		caCert, err := ReadCertFromFile(tlsCaCertFile)
+		if err != nil {
+			return nil, err
+		}
+		if ok := newRootCAs.AppendCertsFromPEM(caCert); !ok {
+			return nil, err
```

Comment:
What's this `err`? It's probably `nil` when we reach this line, but is that what we want? I'd propose something like
```go
return nil, fmt.Errorf("could not append CA cert from %q", tlsCaCertFile)
```